src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/Reload/ProjectReloadInterceptor.cs

```diff
@@ -1,62 +1,62 @@
 ﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
 
 using System;
+using System.Collections.Generic;
 using System.Collections.Immutable;
 using System.ComponentModel.Composition;
+using System.Linq;
 
 using Microsoft.Build.Construction;
 
 namespace Microsoft.VisualStudio.ProjectSystem
 {
     /// <summary>
-    /// C#/VB specific project reload interceptor.
+    ///     Reloads a project if the number of configuration dimensions has changed.
     /// </summary>
     [Export(typeof(IProjectReloadInterceptor))]
-    [AppliesTo(ProjectCapability.DotNet)]
+    [AppliesTo(ProjectCapabilities.ProjectConfigurationsDeclaredDimensions)]
     internal sealed class ProjectReloadInterceptor : IProjectReloadInterceptor
     {
         [ImportingConstructor]
-        public ProjectReloadInterceptor()
+        public ProjectReloadInterceptor(UnconfiguredProject project)
         {
+            DimensionProviders = new OrderPrecedenceImportCollection<IProjectConfigurationDimensionsProvider>(projectCapabilityCheckProvider: project);
         }
 
-        public ProjectReloadResult InterceptProjectReload(ImmutableArray<ProjectPropertyElement> oldProperties, ImmutableArray<ProjectPropertyElement> newProperties)
+        [ImportMany]
+        public OrderPrecedenceImportCollection<IProjectConfigurationDimensionsProvider> DimensionProviders
         {
-            if (NeedsForcedReload(oldProperties, newProperties))
-            {
-                return ProjectReloadResult.NeedsForceReload;
-            }
-
-            return ProjectReloadResult.NoAction;
+            get;
         }
 
-        private static bool NeedsForcedReload(ImmutableArray<ProjectPropertyElement> oldProperties, ImmutableArray<ProjectPropertyElement> newProperties)
+        public ProjectReloadResult InterceptProjectReload(ImmutableArray<ProjectPropertyElement> oldProperties, ImmutableArray<ProjectPropertyElement> newProperties)
         {
-            // If user added or removed TargetFramework/TargetFrameworks property, then force a full project reload.
-            (bool hasTargetFramework, bool hasTargetFrameworks) = ComputeProjectTargets(oldProperties);
-            (bool hasTargetFramework, bool hasTargetFrameworks) newTargets = ComputeProjectTargets(newProperties);
+            IEnumerable<string> oldDimensionsNames = GetDimensionsNames(oldProperties);
+            IEnumerable<string> newDimensionsNames = GetDimensionsNames(newProperties);
```

Comment:
Consider ordering the dimension names before comparing the sequences. A re-order, particularly if the first framework doesn't change, is presumably not a force-reload-worthy event.

Reply:
The order actually matters (its used in the name of the config), and it is currently guaranteed to be [Configuration, Platform, TargetFramework] in that exact order.

Reply:
Thanks for the clarification, I wasn't 100% sure on it.